yaml/_dumper/dumper.ts

```diff
@@ -780,13 +775,15 @@ function writeNode(
           state.dump = `&ref_${duplicateIndex} ${state.dump}`;
         }
       }
-    } else if (type === "[object String]") {
+    } else if (typeof state.dump === "string") {
       if (state.tag !== "?") {
         writeScalar(state, state.dump, level, iskey);
       }
     } else {
       if (state.skipInvalid) return false;
-      throw new YamlError(`unacceptable kind of an object to dump ${type}`);
+      throw new YamlError(
+        `unacceptable kind of an object to dump ${state.dump}`,
```

Comment:
This doesn't show the type of `state.dump` with this change
```suggestion
        `unacceptable kind of an object to dump ${_toString.call(state.dump)}`,
```

Reply:
It should. Can you elaborate?

Reply:
If `state.dump` is, for example, a number, `_toString.call(state.dump)` is `[object Number]`, but `${state.dump}` is something like `42` (the number itself)

Reply:
Oh, right. Would `typeof` be better?

Reply:
`typeof` works for most cases, but it doesn't work for `null` and arrays (both becomes `object`)

Reply:
Ah, true. Will revert.

Reply:
Actually, `Object.prototype.toString(null)` and `Object.prototype.toString([])` both evaluate to `[object Object]`. I'll go with `typeof` as it's a little cleaner.

Reply:
> Actually, Object.prototype.toString(null) and Object.prototype.toString([]) both evaluate to [object Object]

No, existing code is equivalent of `Object.prototype.toString.call(null)` and `Object.prototype.toString.call([])`. They evaluate to `[object Null]` and `[object Array]`

Reply:
Ok. Thanks for clarifying. TIL.